modules/geoip2/geoip-parser.c

```diff
@@ -75,7 +75,16 @@ _mmdb_load_entry_data_list(GeoIPParser *self, const gchar *input, MMDB_entry_dat
 
   if (!result.found_entry)
     {
-      mmdb_problem_to_error(_gai_error, mmdb_error, "lookup");
+      if (0 != _gai_error)
```

Comment:
Are there any reasons to use this pattern in conditional expressions?
In good old days, this kind of pattern was used to save your variable from an unwanted assignment... (eg.: `_gai_error = 0` vs. `_gai_error == 0`) but nowadays compilers detects this situation (-Wparentheses maybe).
And in Java there is a similar pattern when you write this:
```
"mystring".equalsTo(object)
```
instead of checking null pointers, etc...

In syslog-ng we usually use the other way, variable first, then literal/constant... and it's more readable (at least for me...).

Reply:
Sorry, I see that you just keep the original code... then we will discuss this pattern IRL on next handover :)

Reply:
Yeah. I stopped the refactor there, but sure I'll update accordingly.

Note: Personally I like yoda condition, but do not have hard feeling for it.

Reply:
Yoda condition... so they also gave a name for this pattern. Thanks :)